### PR TITLE
Add full intent name to ranked responses

### PR DESCRIPTION
**Pull request by hotzenklotz**
**Proposed changes**:
- When using a `ResponseSelector` a user action has access to the `full_retrieval_intent` property for the predicted response. Additionally, the `ResponseSelector` provides a `ranking` of all predicted responses with only the properties ´name` and `confidence`.

- For downstream tasks it would be useful to have the `full_retrieval_intent` property also available for the `ranked` responses. This PR adds this feature.

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by sara-tagger** [comment]
Thanks for submitting a pull request 🚀 @melindaloubser1 will take a look at it as soon as possible ✨

File: changelog/5690.improvement.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Add `full_retrieval_intent` property to `ResponseSelector` rankings
```
Comment (tmbo) [inline changelog/5690.improvement.rst:1]:
```suggestion
Add ``full_retrieval_intent`` property to ``ResponseSelector`` rankings
```

**Comment by hotzenklotz** [comment]
@tmbo I extended a test to cover the `ranking` property and make sure it has a `full_retrieval_intent` property.

File: tests/nlu/selectors/test_selectors.py [hunk 2]
```diff
@@ -43,3 +43,11 @@ def test_train_selector(pipeline, component_builder, tmpdir):
         .get("default")
         .get("full_retrieval_intent")
     ) is not None
+
+    ranking = parsed.get(RESPONSE_SELECTOR_PROPERTY_NAME).get("default").get("ranking")
+    assert (ranking) is not None
```
Comment (tmbo) [inline tests/nlu/selectors/test_selectors.py:48]:
```suggestion
    assert ranking is not None
```

**Pull request merged**